esp/platform/persistent.cpp

```diff
@@ -140,6 +151,18 @@ class CAvailKeeper
         }
         return nullptr;
     }
+    void cleanup()
+    {
+        HashIterator iter(m_avail4ep);
+        std::vector<EpSocketSetMapping*> elems;
+        for (iter.first(); iter.isValid(); iter.next())
+        {
+            if ((*m_avail4ep.mapToValue(&iter.query()))->empty())
+                elems.push_back(static_cast<EpSocketSetMapping*>(&iter.query()));
+        }
```

Comment:
the Map should be iterator range compatible, so I think you express the above loop without the HashIterator, more simply as:
```
    for (auto &e: m_avail4ep)
    {
        if (e.getValue()->empty())
            elems.push_back(&e);
    }
```